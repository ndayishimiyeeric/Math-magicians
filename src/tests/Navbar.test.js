import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router, Link } from 'react-router-dom';

test('renders correctly', () => {
  const tree = TestRenderer
    .create(<Router><Link to="/">Home</Link></Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('render Links', () => {
  test('Home Link', () => {
    const tree = TestRenderer
      .create(<Router><Link to="/">Home</Link></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Calculator Link', () => {
    const tree = TestRenderer
      .create(<Router><Link to="/">Calculator</Link></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Quote Link', () => {
    const tree = TestRenderer
      .create(<Router><Link to="/quote">Quote</Link></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
