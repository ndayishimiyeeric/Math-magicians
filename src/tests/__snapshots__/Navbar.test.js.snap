// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Links Calculator Link 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  Calculator
</a>
`;

exports[`render Links Home Link 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  Home
</a>
`;

exports[`render Links Quote Link 1`] = `
<a
  href="/quote"
  onClick={[Function]}
>
  Quote
</a>
`;

exports[`renders correctly 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  Home
</a>
`;
